### PR TITLE
Backup unconnected persona

### LINKED ISSUE #965: [Bug] Cannot backup Persona through QR if no SNS connected
# Bug Report

Screen Shot 2020-04-10 at 15 00 43

### System

- OS:
- OS Version:

### Browser

- Browser:Firefox
- Browser Version:

### Maskbook

- Maskbook Version:1.10.11
- Installation: /* May be "Store", "ZIP", or "Self-Complied" */
- Build Commit: Optionally attach a Commit ID, if it is from an pre-release branch head

### Expected Behavior

/* Write the expected behavior here. */

### Actual Behavior

/* Write the actual behavior here. */

### How To Reproduce

Create a new Persona such as '123' pw '123'.
Try to backup Persona through QR -> cant render QR code
/* Specify how it may be produced here. */

### PR DESCRIPTION
**Pull request by guanbinrui**
close #965

### CODE REVIEW

File: src/_locales/zh/messages.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "back": "返回",
     "backup": "備份",
     "backup_persona": "備份角色",
+    "backup_persona_unavailable": "請先將角色連接到社交網絡。",
```
Comment (septs) [inline src/_locales/zh/messages.json:20]: require "camera unavailable", the case handle.

File: src/_locales/en/messages.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
     "back": "Back",
     "backup": "Backup",
     "backup_persona": "Backup your persona.",
+    "backup_persona_unavailable": "Please connect persona with a social network.",
```
Comment (septs) [inline src/_locales/en/messages.json:20]:
```
    "camera_unavailable": "Your camera is not available.",
```

Comment (guanbinrui) [reply src/_locales/en/messages.json:20]: This isn't a message when users wanna backup an unconnected persona. The camera only used when restoring persona.

**Comment by guanbinrui** [comment]
@neruthes Can you accept an error message as the preview image?

File: src/_locales/en/messages.json (hunk 2 above)
Comment (septs) [reply src/_locales/en/messages.json:20]: ACK

File: src/_locales/zh/messages.json (hunk 1 above)
Comment (Tedko) [inline src/_locales/zh/messages.json:20]: 「請先綁定社交網絡」
BTW why use 角色

Comment (guanbinrui) [reply src/_locales/zh/messages.json:20]: It's everywhere.

**Comment by neruthes** [comment]
> @neruthes What do you think about an error message in the preview image?

What error message exactly?

As I understand it, since this PR, a Persona without profiles can be backed up properly. There will hardly be any error.

**Comment by Tedko** [comment]
“ can be backed up properly” — not for QR

**Comment by guanbinrui** [comment]
@Jack-Works Can we restore a from the compressed backup format which contains persona data only?

**Comment by Jack-Works** [comment]
IIRC no. Compressed format need a profile to do so

**Comment by neruthes** [comment]
Temporary workaround: A profileless Persona is treated as having a profile of `3273a3e1-b81a-4ec1-8853-910caeba76c9` at `example.com`.

**Comment by guanbinrui** [comment]
@neruthes @Jack-Works @Tedko I tweaked the compression methods make it support unconnected persona.

File: src/utils/type-transform/BackupFileShortRepresentation.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ import type { ProfileRecord } from '../../database/Persona/Persona.db'
 
 export type BackupJSONFileLatestShort = [
     string, // version, should be "1"
```
Comment (Jack-Works) [inline src/utils/type-transform/BackupFileShortRepresentation.ts:7]: Notice: if you changed the format, you should change the version too

Comment (guanbinrui) [reply src/utils/type-transform/BackupFileShortRepresentation.ts:7]: Actually, the format of the backup data didn't change. It's kinda bugfix. The user is able to restore the persona with the QR code which generates before these changes.

**Comment by Jack-Works** [comment]
@guanbinrui  hi there is merge conflict

**Comment by Jack-Works** [comment]
let me resolve the conflict

**Comment by Tedko** [comment]
looks good to me.

**Pull request merged**